File: cs/Problems/Day2Test.cs
namespace aoc24.Problems;

public class Day2Test
{
    private readonly Day2 sut = new();

    [Theory, InlineData(2)]
    public void TestSet_ShouldYield_Result(int expected)
    {
        var input = InputReader.ReadProblemInput("day2_1");
        var result = sut.Solve(input);

        Assert.Equal(expected, result);
    }

    [Theory, InlineData(486)]
    public void FullSet_ShouldYield_Result(int expected)
    {
        var input = InputReader.ReadProblemInput("day2_2");
        var result = sut.Solve(input);

        Assert.Equal(expected, result);
    }
}

File: go/problems/day2_test.go
package problems

import "testing"

func TestDay2_TestSet_YieldsExpectedResult(t *testing.T) {
	TestAocProblem(t, Day2{}, "day2_1", 2)
}

func TestDay2_FullSet_YieldsExpectedResult(t *testing.T) {
	TestAocProblem(t, Day2{}, "day2_2", 486)
}
